src/network/networkd-ndisc.c

```diff
@@ -950,6 +950,105 @@ static int ndisc_router_process_captive_portal(Link *link, sd_ndisc_router *rt)
         return 1;
 }
 
+static void ndisc_pref64_hash_func(const NDiscPREF64 *x, struct siphash *state) {
+        assert(x);
+
+        siphash24_compress(&x->prefix_len, sizeof(x->prefix_len), state);
+        siphash24_compress(&x->prefix, sizeof(x->prefix), state);
+}
+
+static int ndisc_pref64_compare_func(const NDiscPREF64 *a, const NDiscPREF64 *b) {
+        int r;
+
+        assert(a);
+        assert(b);
+
+        r = CMP(a->prefix_len, b->prefix_len);
+        if (r != 0)
+                return r;
+
+        return  memcmp(&a->prefix, &b->prefix, FAMILY_ADDRESS_SIZE(AF_INET6));
```

Comment:
Doubled space.
`FAMILY_ADDRESS_SIZE()` is redundant, as this is always for IPv6. Please use `sizeof(a->prefix)`.